upup/models/cloudup/resources/addons/node-termination-handler.aws/queue-processor-k8s-1.11.yaml.template

```diff
@@ -0,0 +1,277 @@
+          image: public.ecr.aws/aws-ec2/aws-node-termination-handler:v1.12.1
+          imagePullPolicy: IfNotPresent
+          securityContext:
+            readOnlyRootFilesystem: true
+            runAsNonRoot: true
+            runAsUser: 1000
+            runAsGroup: 1000
+            allowPrivilegeEscalation: false
+          env:
+            - name: NODE_NAME
+              valueFrom:
+                fieldRef:
+                  fieldPath: spec.nodeName
+            - name: POD_NAME
+              valueFrom:
+                fieldRef:
+                  fieldPath: metadata.name
+            - name: NAMESPACE
+              valueFrom:
+                fieldRef:
+                  fieldPath: metadata.namespace
+            - name: SPOT_POD_IP
+              valueFrom:
+                fieldRef:
+                  fieldPath: status.podIP
+            - name: DELETE_LOCAL_DATA
+              value: ""
+            - name: IGNORE_DAEMON_SETS
+              value: ""
+            - name: POD_TERMINATION_GRACE_PERIOD
+              value: ""
+            - name: INSTANCE_METADATA_URL
+              value: ""
+            - name: NODE_TERMINATION_GRACE_PERIOD
+              value: ""
+            - name: WEBHOOK_URL
+              value: ""
+            - name: WEBHOOK_HEADERS
+              value: ""
+            - name: WEBHOOK_TEMPLATE
+              value: ""
+            - name: DRY_RUN
+              value: "false"
+            - name: METADATA_TRIES
+              value: "3"
+            - name: CORDON_ONLY
+              value: "false"
+            - name: TAINT_NODE
+              value: "false"
+            - name: JSON_LOGGING
+              value: "false"
+            - name: LOG_LEVEL
+              value: "info"
+            - name: WEBHOOK_PROXY
+              value: ""
+            - name: ENABLE_PROMETHEUS_SERVER
+              value: "false"
+            - name: ENABLE_SPOT_INTERRUPTION_DRAINING
+              value: "false"
+            - name: ENABLE_SCHEDULED_EVENT_DRAINING
+              value: "false"
+            - name: ENABLE_REBALANCE_MONITORING
+              value: "false"
+            - name: ENABLE_SQS_TERMINATION_DRAINING
+              value: "true"
+            - name: QUEUE_URL
+              value: "{{ DefaultQueueName }}"
+            - name: PROMETHEUS_SERVER_PORT
+              value: "9092"
+            - name: AWS_REGION
+              value: ""
+            - name: AWS_ENDPOINT
+              value: ""
+            - name: CHECK_ASG_TAG_BEFORE_DRAINING
+              value: "true"
+            - name: MANAGED_ASG_TAG
+              value: "aws-node-termination-handler/managed"
+            - name: WORKERS
+              value: "10"
+          resources:
+            limits:
+              cpu: 100m
+              memory: 128Mi
+            requests:
+              cpu: 50m
+              memory: 64Mi
+      nodeSelector:
+        kubernetes.io/os: linux
+      tolerations:
```

Comment:
This is probably the cause of the mentioned SQS errors. This means the pod can randomly be assigned to CP nodes, which are not given the SQS privileges.

Any particular reason for this?

Reply:
Nope, just a miss on my part. 

Thinking about it more, it might make more sense to limit nth queue processor to only CP nodes. And even with that I still think IRSA would be better to use compared to applying the policy to all CP nodes. Thoughts?

Reply:
if you look at the linked PR I have, you will see that we add the privileges to CP by default, but if irsa is available, we add the same policy to the IAM role instead. If you apply the privileges to CP by default as well, and add the appropriate node selector, I can add the irsa variant once that is ready.